api.go

```diff
@@ -189,6 +189,22 @@ type OnboardByOwnerRequest struct {
 	LayoutPosition string `json:"layoutPosition,omitempty"` // pattern: GATEWAY|STANDALONE|ENDNODE, STANDALONE by default
 }
 
+// OnboardEndnodeWithGatewayRequest for requesting Onboard endnode with gateway id.
+type OnboardEndnodeWithGatewayRequest struct {
```

Comment:
**OnboardEndnodeWithGatewayRequest**
naming looks indecisive.

Suggested candidates:
### Fully Qualified:
Prepare 2 structs for the different type of the content-type.
- OnboardEndnodeWithGatewayThingIDRequest
- OnboardEndnodeWithGatewayVendorThingIDRequest

### Pack it in one struct.
- OnboardEndnodeRequest
The only difference between those content-types are gatewayVendorThingID and gatewayThingID.
Only accept either of them. (Somehow.)